online/src/app/edit-product-form/edit-product-form.component.ts

```diff
@@ -13,12 +15,18 @@ import { Location } from '@angular/common'
 export class EditProductFormComponent implements OnInit {
   editForm: FormGroup | undefined;
   product: Product | undefined;
-  constructor(private productService: ProductService, private route: ActivatedRoute, private formBuilder: FormBuilder,
-    private location: Location) { }
+  selectedProduct = this.store.select(selectOneProduct);
+
+  constructor(
+    private route: ActivatedRoute,
+    private formBuilder: FormBuilder,
+    private location: Location,
+    private store: Store<AppState>) { }
 
   ngOnInit(): void {
     const id = this.route.snapshot.params['id'];
-    this.productService.getProductById(id).pipe(take(1)).subscribe((item: Product) => {
+    this.store.dispatch(getProduct({ id: id }));
```

Comment:
if propery and value have the same name, you can specify it only once (e.g., { id } )

Note: this doesn't apply if you have to call it from another scope (e.g., this.blabla or Class.blabla)